core/src/test/scala/org/apache/spark/util/JsonProtocolSuite.scala

```diff
@@ -984,25 +993,40 @@ class JsonProtocolSuite extends SparkFunSuite {
     val gettingResult = SparkListenerTaskGettingResult(taskInfo)
 
     assert(
-      stageSubmittedFromJson(sparkEventToJsonString(stageSubmitted)).stageInfo.accumulables.isEmpty)
+      jsonProtocol.stageSubmittedFromJson(
+        jsonProtocol.sparkEventToJsonString(stageSubmitted)).stageInfo.accumulables.isEmpty)
     assert(
-      taskStartFromJson(sparkEventToJsonString(taskStart)).taskInfo.accumulables.isEmpty)
+      jsonProtocol.taskStartFromJson(
+        jsonProtocol.sparkEventToJsonString(taskStart)).taskInfo.accumulables.isEmpty)
     assert(
-      taskGettingResultFromJson(sparkEventToJsonString(gettingResult))
+      jsonProtocol.taskGettingResultFromJson(
+          jsonProtocol.sparkEventToJsonString(gettingResult))
         .taskInfo.accumulables.isEmpty)
 
     // Deliberately not fixed for job starts because a job might legitimately reference
     // stages that have completed even before the job start event is emitted.
-    testEvent(jobStart, sparkEventToJsonString(jobStart))
+    testEvent(jobStart, jsonProtocol.sparkEventToJsonString(jobStart))
+  }
+
+  test("SPARK-49872: allow to limit json reader string sizes") {
+    val bigStringEvent = SparkListenerExecutorUnexcluded(
+      executorUnexcludedTime, "a".repeat(10_000))
+    val jsonString = jsonProtocol.sparkEventToJsonString(bigStringEvent)
+    assert(jsonProtocol.sparkEventFromJson(jsonString) == bigStringEvent)
+    val jsonProtocolWithLimit = new JsonProtocol(new SparkConf()
+      .set(EVENT_LOG_READER_MAX_STRING_LENGTH, 1_000))
+    assertThrows[com.fasterxml.jackson.core.JsonProcessingException] {
```

Comment:
It is recommended to add assertions for the content of the error message, for example:

```
val ex = intercept[JsonProcessingException] {
      jsonProtocolWithLimit.sparkEventFromJson(jsonString)
    }
    assert(ex.getMessage.contains(s"String value length (${bigStr.length}) exceeds the maximum " +
      s"allowed ($maxStringLengthJson, from `StreamReadConstraints.getMaxStringLength()`)"))
  }
```

Reply:
Done